realm/src/main/java/io/realm/Realm.java

```diff
@@ -194,10 +194,14 @@ public static Realm getInstance(Context context) {
      * model classes or version has has changed so a migration is required.
      */
     public static Realm getDefaultInstance() {
-        if (defaultConfiguration == null) {
+        final RealmConfiguration configuration;
+        synchronized (Realm.class) {
+            configuration = defaultConfiguration;
+        }
```

Comment:
@cmelchior Hm... As you said, I don't have any real scenario.

Is it enough just to introduce local variable to avoid accidental NPE in `create()`?

```
    public static Realm getDefaultInstance() {
        final RealmConfiguration configuration = defaultConfiguration;
        if (configuration == null) {
            throw new NullPointerException("No default RealmConfiguration was found. Call setDefaultConfiguration() first");
        }
        return create(configuration);
    }
```